bluesky/tests/test_examples.py

```diff
@@ -166,58 +154,68 @@ def print_event_time(name, doc):
 
 
 def test_calltime_subscription():
-    assert_equal(RE.state, 'idle')
+    assert RE.state == 'idle'
     RE(simple_scan_saving(det, motor), subs={'event': print_event_time})
-    assert_equal(RE.state, 'idle')
+    assert RE.state == 'idle'
 
 
 def test_stateful_subscription():
-    assert_equal(RE.state, 'idle')
+    assert RE.state == 'idle'
     token = RE.subscribe('event', print_event_time)
     RE(simple_scan_saving(det, motor))
     RE.unsubscribe(token)
-    assert_equal(RE.state, 'idle')
-
+    assert RE.state == 'idle'
 
 def test_live_plotter():
-    if skip_mpl:
-        raise nose.SkipTest("matplotlib is not available")
+    try:
+        import matplotlib.pyplot as plt
+        del plt
+    except ImportError as ie:
+        pytest.skip("Skipping live plot test because matplotlib is not installed."
+                    "Error was: {}".format(ie))
+
     my_plotter = LivePlot('det', 'motor')
-    assert_equal(RE.state, 'idle')
+    assert RE.state == 'idle'
     RE(stepscan(det, motor), subs={'all': my_plotter})
-    assert_equal(RE.state, 'idle')
+    assert RE.state == 'idle'
 
 
 def test_sample_md_dict_requirement():
     # We avoid a json ValidationError and make a user-friendly ValueError.
-    assert_raises(ValueError, RE, simple_scan(motor), sample=1)
+    with pytest.raises(ValueError):
+        RE(simple_scan(motor), sample=1)
     RE(simple_scan(motor), sample={'number': 1})  # should not raise
 
 
 def test_md_dict():
     yield _md, {}
 
-
 def test_md_historydict():
     try:
-        from historydict import HistoryDict
-    except ImportError:
-        raise SkipTest()
-    yield _md, HistoryDict(':memory:')
+        import historydict
+    except ImportError as ie:
+        pytest.skip('Skipping test because historydict cannot be imported. '
+                    'Error was {}'.foramt(ie))
+    yield _md, historydict.HistoryDict(':memory:')
 
 
 def _md(md):
     RE = RunEngine(md)
     RE.ignore_callback_exceptions = False
     scan = simple_scan(motor)
-    assert_raises(KeyError, RE, scan)  # missing owner, beamline_id
+    with pytest.raises(KeyError):
+        RE(scan)  # missing owner, beamline_id
     scan = simple_scan(motor)
-    assert_raises(KeyError, RE, scan, owner='dan')
+    with pytest.raises(KeyError):
+        RE(scan, owner='dan')
     scan = simple_scan(motor)
     RE(scan, owner='dan', beamline_id='his desk',
        group='some group', config={})  # this should work
     scan = simple_scan(motor)
-    assert_raises(KeyError, RE, scan)  # this should fail; none was persisted
+    with pytest.raises(KeyError):
+        # TODO: Is this persistent thing still true?  I thought that something
```

Comment:
Yes that is exactly what I was thinking of.  Great, thanks.  Removing this comment in a future commit